### DISCUSSION TITLE
Api ideation [Really long term]

### DISCUSSION THREAD

**Discussion by rohanbatrain** [RafaelGB/obsidian-db-folder#650] [Ideas]
I was thinking about dbfolder having an api from a quite long time now, i am not updated with recent changes and changed my account from @rohanbatraio to @rohanbatrain but here's an idea i want to discuss with you @RafaelGB. 

Why not integrate airtable with your existing db plugin? in that way we can use airtable integration with 100+ apps without reinventing the wheel? 

Or even better integrate n8n.io if you really want to kickstart the api integration.

I would love to hear your views.

**Comment by RafaelGB** [comment]
yeah, the API feature could be the next mayor version. The idea I was working on is to expose different implementations of read and write.

the plugin is reaching a point of maturity to start considering it.

The idea of create your own column types, integration with Notion directly (or airtable) and a lot of new possibilities...

**Reply by rohanbatrain** [reply]
I would highly stress on looking into [n8n](https://n8n.io/integrations/notion-beta/). I know you would want to make native notion integration to have complete control but integrating n8n will give the community endless possibilities.

**Reply by rohanbatrain** [reply]
or simply focus on making your api, and then with the endpoints you expose i (or the community) would connect your app + n8n together. That would be hella awesome.

**Reply by RafaelGB** [reply]
with an input/output we can integrate whatever we want, just we have to make the architecture complete enough

**Reply by rohanbatrain** [reply]
Let's see what the future of DBFOLDER unveils. :smile:

**Reply by RafaelGB** [reply]
Commit 41480453

This is the first step for a Input / Output API

**Comment by rohanbatrain** [comment]
idea 2/2

A beneficial concept, not directly related to the plugin, is to create a web application that exposes the entire stack of dbfolder. This would allow rendering with react table to be accessible via a standard web browser on any device connected to the LAN network. With this capability, dbfolder can serve as a source for ObsidianMD, enabling access from any location without requiring synchronization or additional configuration. Simply open a web browser and you're ready to go.

**Reply by rohanbatrain** [reply]
PC running OBSIDIAN + DBFOLDER would act as a "server" and any client running a web browser and accessing the interface is well a "client"

**Reply by RafaelGB** [reply]
Once we define the API with different implementations we can obtain the information from any source.

I was working on this solution with an Abstraction of the view 

https://github.com/RafaelGB/obsidian-db-folder/blob/master/src/views/DatabaseView.tsx

About to create a server could be tricky but I am open to any idea

**Reply by rohanbatrain** [reply]
I understand your point, and it's not necessary to create a server from scratch. The solution would be to integrate dataview.js with your dbparser and then utilize API methods like POST and FETCH to connect with the frontend web application, which runs react tables. The challenge lies in integrating dataview.js with the input and output of your API.

To summarize, the process involves linking Obsidian dataview to the dbfolder parser, which connects to the API using POST and FETCH methods. You mentioned that you are already working on these methods. The react-table parser used in the legacy dbfolder can then be bundled with webpack to function like a web application.

**Reply by rohanbatrain** [reply]
Your API would run somewhere, That somewhere will act as our server.
